lib/modules/datasource/docker/index.ts

```diff
@@ -159,7 +158,11 @@ export async function getAuthHeaders(
       scope = authenticateHeader.params.scope;
     }
 
-    const authUrl = `${authenticateHeader.params.realm}?service=${authenticateHeader.params.service}&scope=${scope}`;
+    let service = authenticateHeader.params.service;
+    if (!is.string(service)) {
+      service = '';
+    }
```

Comment:
nit:

```suggestion
    const service = authenticateHeader.params.service ?? '';
```

But I think it's better to have it as a constant. What do you think?

Reply:
I had it as that originally, but the [linter](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/restrict-template-expressions.md) was complaining that `params.service` could be a `string` or `string[]`:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/auth-header/index.d.ts#L9

Happy to change it, whichever is best.